src/diffusers/loaders/single_file_utils.py

```diff
@@ -2198,3 +2204,250 @@ def swap_scale_shift(weight):
     )
 
     return converted_state_dict
+
+
+def create_autoencoder_dc_config_from_original(original_config, checkpoint, **kwargs):
```

Comment:
I think for new single file models let's not rely on the original configs anymore. This was for legacy support for the SD1.5/XL models with yaml configs. It's better to infer the diffusers config from the checkpoint and use that for loading.

Reply:
This might be a little difficult here, so please lmk if you have any suggestions on what to do.

Some DCAE checkpoints have the exact same structure and configuration, except for `scaling_factor`. For example, `[dc-ae-f128c512-in-1.0-diffusers](https://huggingface.co/mit-han-lab/dc-ae-f128c512-in-1.0-diffusers)` and `[dc-ae-f128c512-mix-1.0-diffusers](https://huggingface.co/mit-han-lab/dc-ae-f128c512-mix-1.0-diffusers)` only differ in their scaling factor.

I'm unsure how we would determine this just by the model structure. Do we rely on the user passing it as a config correctly, and document this info somewhere?

Reply:
I think that's fine since in the snippet in the docs, we're doing the same thing just with `original_config` instead of `config` right?

Reply:
Updated usage to `config` now and verified that it works. Thank you for the fixes and suggestions!